### ISSUE TITLE
Feature Request: Personality that forces a ship to stay very close to the star

### ISSUE THREAD

**Issue by Bladewood** [endless-sky/endless-sky#2883]
For solar panels and/or ramscoops.

A hypothetical plant race, for example, might focus heavily on solar panel technology (for obvious reasons), and would thus stay very close to the star.

**Comment by tehhowch** [comment]
The only issue i see is that ships with this personality would either never pursue a missile boat that was attacking them, or actually would pursue the missile boat and end up stranded.


I might classify this as needing to wait until we have some more aware AI movement routines, as right now the only position check is "is my target going to cross the constrainment boundary?". Everything else is basically CHASE HOSTILE or JUMP AWAY or LAND THERE.

We'd want a ship with this personality to prefer movement actions that take it closer to the star, but not disallow movement actions that take it away (if it needs to land, for example, or to fight a hostile that is a bit further out, but not too far out). Distance to star is essentially an aura weapon, and could build off logic created to govern NPC movement in the presence of various auras.
As a reminder, @Hadron1776  and I have already written an working [aura class](https://github.com/endless-sky/endless-sky/compare/master...Hadron1776:jump-interdiction), but it doesn't handle any sort of damage.

**Comment by Amazinite** [comment]
(Kind of off topic, but what about this: an aura weapon that creates solar wind to power nearby ships with solar panels, thus creating a perfect situation for ships to use the swarming personality, for they would actually gain something from swarming around a ship with such a weapon. It would allow solar panel powered ship to travel anywhere, assuming the aura ship is alive.)

**Comment by KyroFirehammer** [comment]
Is it just me or do the quarg do this already for ramscooping in kor space

**Comment by jafdy** [comment]
No, I don't think so. But then I rarely know what your on about, so I could have misunderstood.

**Comment by MessyMix** [comment]
Well, think I may have understood him. They may or may not stay in the system for ramscoop, but I've seen this behavior before. IIRC it was a bug that's been fixed.

https://github.com/endless-sky/endless-sky/issues/2312 may help.

**Comment by KiLEdEnNis** [comment]
he meant that Quarg stay near the system center to ramscoop
no; they jump from sector to sector, if there's no enemies they have no reason to go away from the center of said sector
only in actual fights the Quarg leave the system center, just like every ship ever

**Comment by tehhowch** [comment]
If a ship is out of fuel and cannot land / ask for help, it will go park on the star until it gets enough fuel to jump. This Feature Request is for a ship personality that will bias its movement options (e.g. when fighting) to prefer staying near to the star, rather than chasing off into the darkness where solar generation is pitifully low, and the ship might get stranded due to low energy production.

**Comment by Amazinite** [comment]
Right now, systems have a default system fence of 10,000 units; ships that fly beyond that range will attempt to return to being within that range from the system center. This system fence can be customized on a per-system basis.

In order to implement this idea, we could have a customizable system fence on a per-fleet basis. That is, instead of having every fleet in the system adhere to the same system fence range, certain fleets could have restricted ranges that force them to stay close to the system center.
